src/connectionController.ts

```diff
@@ -106,11 +106,15 @@ export default class ConnectionController {
   // Used by other parts of the extension that respond to changes in the connections.
   private eventEmitter: EventEmitter = new EventEmitter();
 
-  constructor(
-    statusView: StatusView,
-    storageController: StorageController,
-    telemetryService: TelemetryService
-  ) {
+  constructor({
+    statusView,
+    storageController,
+    telemetryService,
+  }: {
+    statusView: StatusView;
+    storageController: StorageController;
+    telemetryService: TelemetryService;
+  }) {
     this._statusView = statusView;
     this._storageController = storageController;
     this._telemetryService = telemetryService;
```

Comment:
Just FYI there is a pattern in TS if your constructors are just assigning stuff to instance properties:

```suggestion
  constructor(
    private _statusView: StatusView;
    private _storageController: StorageController;
    private _telemetryService: TelemetryService;
  ) {
```

Basically depends on what you are optimizing for: constructor params changing all the time or constructor just storing passed params (but yeah, this doesn't work with descructuring)

Reply:
I think we add new parameters to constructors not so often, but when we do yeah this requires quite a work to refactor all controllers and tests. Don't have a strong opinion here.

Reply:
I think we want the flexibility of moving things around that destructuring gives here. For instance, down the line we might want to move some of these parts, like the telemetry service and storage controller, to a global context for the extension.